frontend/src/components/ViewTrips/trips-container.js

```diff
@@ -0,0 +1,88 @@
+import React from 'react';
+import Trip from './trip';
+
+import * as DATABASE from '../../constants/database.js';
+
+
+/**
+ * Returns a promise of a query object containg the array of Trip Documents
+ * corresponding to the trips that the current user is a collaborator on.
+ *
+ * @param {firebase.firestore.Firestore} db The Firestore database instance.
+ * @param {string} userEmail The email corresponding to the current user
+ *    logged in.
+ * @return {Promise<!firebase.firestore.QuerySnapshot>} Promise object
+ *    containing the query results with zero or more Trip  documents.
+ */
+function queryUserTrips(db, userEmail) {
+  return db.collection(DATABASE.COLLECTION_TRIPS)
+      .where(DATABASE.TRIPS_COLLABORATORS, 'array-contains', userEmail)
+      .get();
+}
+
+/**
+ * Grabs Trips query result from `queryUserTrips()` and returns an array of
+ * `<Trip>` elements as defined in `trip.js`.
+ *
+ * @param {Promise<!firebase.firestore.QuerySnapshot>} querySnapshot Promise
+ *    object containing the query results with zero or more Trip  documents.
+ * @return {Promise<!Array<Trip>>} Promise object containing an array
+ *    of Trip React/HTML elements corresponding to the Trip docsuments included
+ *    in 'querySnapshot`.
+ */
+function serveTrips(querySnapshot) {
+  return new Promise(function(resolve) {
+    const tripsContainer = querySnapshot.docs.map(doc =>
+        ( <Trip key={doc.id} tripObj={doc.data()} tripId={doc.id} /> ));
```

Comment:
Is it possible for any of these fields to be null? If so, what happens?

Reply:
I believe that this this question has the same answer as the one above:

> Short answer: no.
Long answer: If the props get passed to the Trip component at all, that means a document does exist and at a minimum contains a collaborators field with one element in the array. So it will have a non-null/undefined document id and data.

That being said I don't think I need any other protections here. But if you disagree please let me know.